### ISSUE TITLE
Extract PeerID from multiaddress in HOPR channels smart contract 

### ISSUE THREAD

**Issue by MartinBenediktBusch** [hoprnet/hopr-devrel#318]
### Issue deadline

Feb 1 2023

### Detailed Description
Extract (decode) the PeerID of a HOPR node runner from the announcement event of the HOPRChannels smart contract (0xfabee463f31e39ec8952bbfb4490c41103bf573e). 

We need this in order to link HOPRChannel events (i.e. the creation of a new payment channel) to the staking address of a node using the peerID. This link is important in order to airdrop rewards to node runners based on activities tracked in the HOPRChannels smart contract.

**Comment by robertkiel** [comment]
@MartinBenediktBusch could you describe how you need that? Like how you would interact with that piece of code?

**Comment by MartinBenediktBusch** [comment]
Hi, 

so I need to link staking addresses of people who staked their NR NFT in season 6 to peerIDs and HOPRChannel events (e.g. channel oppening events). I already created a link between peerIDs and staking addresses using the HOPRStaking contract and the NetworkRegistry contract. 

I need somehow to link an event in the HOPR Channels SC to either the PeerID or the staking address. To my understanding one can extract the peerID somehow from the multiaddress from the Announcement Event. 

So basically, I need a tool that extracts the peerID from the multiaddress so that I can link the the 3 Smart contracts.
